### DISCUSSION TITLE
Rule book on aggregation/ Aggregation and disaggregation

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#492] [Ideas]
Hi folks,
an aspect that came to my mind is the way we aggregate/disaggregate equations can sometimes be a bit arbitrary/ we might want to support more in the future.
For example, for the capacity constraint, we use the lowest resolution of all involved flow variables, and aggregate/ disaggregate the units_on variable. 
However, I guess there might be cases, where the user wants to e.g. constrain the unit_capacity by the highest resolution flow variable. A similar example would be the fix- ratio-out-in constraints etc.; again the user might want to define the ratio on the highest involved flow resolution.

The question is whether we want to do something about it. One idea could e.g. be to have a tag when entering the parameter, which allows the user to specify whether this parameter will be used in an aggregating/disaggregating manner. Within SpineOpt, we accordingly decide if we're going to take the lowest or highest resolution for creating the constraint.

There are of course special cases, e.g., the unit_capacity constraint, where more than one parameter (and more than one variable type) is involved. For example, in those cases, we could think of a 3rd option "follow" (or something), e.g., suitable for the `units_on_coeff` in the `fix_ratio_constraint`, which will either aggregate, in case the other variables are of lower resolution or disaggregate in case the other variables are of higher resolution.

@Spine-project/spineopt-developers

**Comment by clizbe** [comment]
@mihlema I think we addressed this today with Germán's multiple resolution discussion (and power vs energy). Or is this something else?
